# Actividad de clase con el dataset Movielens
### EDA


Algunas preguntas que se pueden responder (pero no limitadas a estas)
- Cuáles fueron los géneros mejor rankeados en promedio en el año 2005?
- Cuál es el top 5 de películas mejor rankeadas en promedio?
- Cuál es el género con más revies?
- Cuáles es el género por año con mejor rating entre el año 2000 y 2010?
- Cuál es el usuario con más críticas y cuál es su género favorito?

In [14]:
import pandas as pd

movies = pd.read_csv('data/movies.csv')
ratings = pd.read_csv('data/ratings.csv')
tags = pd.read_csv('data/tags.csv')
links = pd.read_csv('data/links.csv')
movies.head()

movieId                               title  \
0        1                    Toy Story (1995)   
1        2                      Jumanji (1995)   
2        3             Grumpier Old Men (1995)   
3        4            Waiting to Exhale (1995)   
4        5  Father of the Bride Part II (1995)   

                                        genres  
0  Adventure|Animation|Children|Comedy|Fantasy  
1                   Adventure|Children|Fantasy  
2                               Comedy|Romance  
3                         Comedy|Drama|Romance  
4                                       Comedy

In [15]:
movies['year'] = movies['title'].apply(lambda x: x.split('(')[-1][:-1])
movies

movieId                                      title  \
0           1                           Toy Story (1995)   
1           2                             Jumanji (1995)   
2           3                    Grumpier Old Men (1995)   
3           4                   Waiting to Exhale (1995)   
4           5         Father of the Bride Part II (1995)   
...       ...                                        ...   
9737   193581  Black Butler: Book of the Atlantic (2017)   
9738   193583               No Game No Life: Zero (2017)   
9739   193585                               Flint (2017)   
9740   193587        Bungo Stray Dogs: Dead Apple (2018)   
9741   193609        Andrew Dice Clay: Dice Rules (1991)   

                                           genres  year  
0     Adventure|Animation|Children|Comedy|Fantasy  1995  
1                      Adventure|Children|Fantasy  1995  
2                                  Comedy|Romance  1995  
3                            Comedy|Drama|Romance  1995  
4                                          Comedy  1995  
...                                           ...   ...  
9737              Action|Animation|Comedy|Fantasy  2017  
9738                     Animation|Comedy|Fantasy  2017  
9739                                        Drama  2017  
9740                             Action|Animation  2018  
9741                                       Comedy  1991  

[9742 rows x 4 columns]

In [17]:
movies[movies['year'] == '2005']

movieId                                     title  \
5665    27618                Sound of Thunder, A (2005)   
5699    27788                        Jacket, The (2005)   
5709    27821                   Interpreter, The (2005)   
5718    27839                      Ring Two, The (2005)   
5735    30793  Charlie and the Chocolate Factory (2005)   
...       ...                                       ...   
9342   160836                             Hazard (2005)   
9401   164707                          Go Figure (2005)   
9450   167706            Shakespeare Behind Bars (2005)   
9655   180777         Die Frauen von Ravensbrück (2005)   
9691   184349                        Elsa & Fred (2005)   

                                      genres  year  
5665  Action|Adventure|Drama|Sci-Fi|Thriller  2005  
5699           Drama|Mystery|Sci-Fi|Thriller  2005  
5709                          Drama|Thriller  2005  
5718           Drama|Horror|Mystery|Thriller  2005  
5735  Adventure|Children|Comedy|Fantasy|IMAX  2005  
...                                      ...   ...  
9342                   Action|Drama|Thriller  2005  
9401                   Children|Comedy|Drama  2005  
9450                             Documentary  2005  
9655                             Documentary  2005  
9691                    Comedy|Drama|Romance  2005  

[273 rows x 4 columns]

In [ ]:
ratings['userId'].value_counts()

userId
414    2698
599    2478
474    2108
448    1864
274    1346
       ... 
442      20
278      20
147      20
320      20
53       20
Name: count, Length: 610, dtype: int64

El usuario con la mayor cantidad de criticas es el *414* con un total de 2698.